include/swift/AST/Decl.h

```diff
@@ -4465,6 +4465,24 @@ class AbstractStorageDecl : public ValueDecl {
     return {};
   }
 
+  /// \returns the getter decl iff this decl has only one accessor, a 'get'
+  ///          that has an effect (i.e., 'async', 'throws', or both).
+  ///          Otherwise returns nullptr.
+  AccessorDecl *getEffectfulGetAccessor() const;
+
+  /// Performs a "limit check" on an effect possibly exhibited by this storage
+  /// decl with respect to some other storage decl that serves as the "limit."
+  /// This check says that \c this is less effectful than \c other if
+  /// \c this either does not exhibit the effect, or if it does, then \c other
+  /// also exhibits the effect.
+  ///
+  /// \param kind the single effect we are performing a check for.
+  ///
+  /// \returns true iff \c this decl has either does not exhibit the effect,
+  ///          or \c other also exhibits the effect.
+  bool isLessEffectfulThan(AbstractStorageDecl const* other,
+                           EffectKind kind) const;
```

Comment:
:)

Reply:
I just improved this poorly worded comment too :D